src/core/server/models/user/user.ts

```diff
@@ -2323,3 +2351,78 @@ export async function retrieveUserScheduledForDeletion(
   );
   return result.value || null;
 }
+
+/**
+ * createModeratorNote will add a note to a users account
+ * @param mongo the database to put the notification digests into
+ * @param tenantID the ID of the Tenant that this User exists on
+ * @param id the ID of the User who is the subject of the note
+ * @param createdBy the ID of Moderator that is creating the note
+ * @param note the contents of the note
+ * @param now the current time that the note was created
+ */
+export async function createModeratorNote(
+  mongo: Db,
+  tenantID: string,
+  id: string,
+  createdBy: string,
+  note: string,
+  now = new Date()
+) {
+  const moderatorNote: ModeratorNote = {
+    id: uuid(),
+    createdAt: now,
+    body: note,
+    createdBy,
+  };
+  const result = await collection(mongo).findOneAndUpdate(
+    { id, tenantID },
+    {
+      $push: {
+        moderatorNotes: moderatorNote,
+      },
+    },
+    {
+      // False to return the updated document instead of the original
+      // document.
+      returnOriginal: false,
+    }
+  );
+  if (!result.value) {
+    throw new UserNotFoundError(id);
+  }
+
+  return result.value;
+}
+
+/**
+ * deleteModeratorNote will remove a note from a user profile
+ * @param mongo the database to put the notification digests into
+ * @param tenantID the ID of the Tenant that this User exists on
+ * @param userID the ID of the user
+ * @param id the ID of the note to delete
+ * @param createdBy the ID of the note author
+ */
+export async function deleteModeratorNote(
+  mongo: Db,
+  tenantID: string,
+  userID: string,
+  id: string,
+  createdBy: string
+) {
+  const result = await collection(mongo).findOneAndUpdate(
+    {
+      id: userID,
+      tenantID,
+    },
+    {
+      $pull: {
+        moderatorNotes: { id, createdBy },
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
```

Comment:
Probably good to add the guard clause from the `createModeratorNote` function that returns the `UserNotFoundError` on the falsy response of `findOneAndUpdate`.